ADR/0030-tekton-results-naming-convention.md

```diff
@@ -0,0 +1,257 @@
+    },
+    "successes": {
+      "type": "integer",
+      "minimum": 0
+    },
+    "note": {
+      "type": "string",
+    },
+    "failures": {
+      "type": "integer",
+      "minimum": 0
+    },
+    "warnings": {
+      "type": "integer",
+      "minimum": 0
+    }
+  },
+  "required": ["result", "timestamp", "successes", "failures", "warnings"]
+}
+```
+
+### Results for Scan-Like Tasks
+Scan-like tasks are those that search for issues within the code base or artifact whose results raise concerns (vulnerabilities) according to some internal severity or classification metric. These results are expressly different from test-like results as there may need to be additional criteria applied to the results to determine what vulnerabilities are permissable for the final artifact.
+
+Scan-like results should still use the **TEST_OUTPUT** result for indicating whether the scan has run successfully or if it has errored, but this output should not indicate a failure.
+
+To display count of found vulnerabilities and make it easy to understand and evaluate the state of scanned content, the additional output of scan-like tasks will be provided in a minimized [Tekton result](https://tekton.dev/docs/pipelines/tasks/#emitting-results) in JSON format listing. The name of the result will be **SCAN_OUTPUT**.
+
+While the vulnerability classifications should remain consistent in order to enable easier extensions into other AppStudio components (namely the user interface and enterprise contract), it is the responsibility of every scan-like task to inform the user about what criteria fits each vulnerability classification used. The vulnerability classifications from most important/severe to least are **critical**, **high**, **medium**, and **low**. If a vulnerability is classified as **unknown** then the scanner cannot make further judgement about its severity.
+
+Some scanners are additionally aware of whether a specific vulnerability is patched or unpatched (i.e. whether there is a known fix that has been published by the vulnerable package's maintainers). If vulnerabilities are known to be unpatched, the scanner may use the **unpatched_vulnerabilities** object to represent their quantities and severities.
+
+#### Tekton Result Format for `SCAN_OUTPUT`
+The output of the Tekton result **SCAN_OUTPUT** will be a JSON object that includes the following information about the found vulnerabilities. While the vulnerability classification may vary based on the scanner used in the task, an example description of vulnerability severity ratings can be found at the [Red Hat Vulnerability documentation](https://access.redhat.com/articles/red_hat_vulnerability_tutorial).
+
+```
+{
+  "vulnerabilities": {
+    "critical": 1,
+    "high": 0,
+    "medium": 1,
+    "low":0,
+    "unknown":0
+  },
+  "unpatched_vulnerabilities": {
+    "critical": 0,
+    "high": 1,
+    "medium": 0,
+    "low":1
```

Comment:
This misses `unknown`

Reply:
`unknown` is optional for backwards compatibility and was therefore intentionally not included